File: Easy/100-same-tree/try-3.py
/*
 * Problem #100: Same Tree
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 01/01/2025, 17:49:10
 * Link: https://leetcode.com/problems/same-tree/
 */

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        dq_p = deque([p])
        dq_q = deque([q])
        while dq_p and dq_q :
            p1 = dq_p.popleft()
            q1 = dq_q.popleft()
            print(p1,q1)
            if p1 == None and q1 == None:
                continue
            elif (p1 == None or q1 == None) or ( p1.val != q1.val ):
                return False
        
            dq_p.append(p1.left)
            dq_p.append(p1.right)
            dq_q.append(q1.left)
            dq_q.append(q1.right)
        return True

File: Easy/104-maximum-depth-of-binary-tree/try-3.cpp
/*
 * Problem #104: Maximum Depth of Binary Tree
 * Difficulty: Easy
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 11/03/2025, 16:48:13
 * Link: https://leetcode.com/problems/maximum-depth-of-binary-tree/
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(root == nullptr){
            return 0;
        }

        deque<TreeNode*> dq;
        int i,currsize,c = 0;
        TreeNode* element;

        dq.push_back(root);
        while(dq.size()){
            currsize = dq.size();
            for(i = 0;i < currsize;i++){
                element = dq.front();
                dq.pop_front();
                if(element->right){
                    dq.push_back(element->right);
                }
                if(element->left){
                    dq.push_back(element->left);
                }
            }
            c++;
        }
        return c;
    }
};
